packages/flutter/lib/src/services/restoration.dart

```diff
@@ -319,7 +319,18 @@ class RestorationManager extends ChangeNotifier {
       return null;
     }
     final ByteData encoded = data.buffer.asByteData(data.offsetInBytes, data.lengthInBytes);
-    return const StandardMessageCodec().decodeMessage(encoded) as Map<Object?, Object?>?;
+    Map<Object?, Object?>? decodedMessage;
+    try {
+      decodedMessage = const StandardMessageCodec().decodeMessage(encoded) as Map<Object?, Object?>?;
```

Comment:
I think this could just be

```suggestion
      return const StandardMessageCodec().decodeMessage(encoded) as Map<Object?, Object?>?;
```

and then the `decodedMessage` variable can be removed.